cmd/runtime_options.go

```diff
@@ -63,24 +67,42 @@ extended: base + Babel with ES2015 preset + core.js v2,
 	return flags
 }
 
-func getRuntimeOptions(flags *pflag.FlagSet) (lib.RuntimeOptions, error) {
+func getRuntimeOptions(flags *pflag.FlagSet, environment map[string]string) (lib.RuntimeOptions, error) {
 	opts := lib.RuntimeOptions{
 		IncludeSystemEnvVars: getNullBool(flags, "include-system-env-vars"),
 		CompatibilityMode:    getNullString(flags, "compatibility-mode"),
 		Env:                  make(map[string]string),
 	}
 
-	// If enabled, gather the actual system environment variables
-	if opts.IncludeSystemEnvVars.Bool {
-		opts.Env = collectEnv()
+	if !opts.CompatibilityMode.Valid { // If not explicitly set via CLI flags, look for an environment variable
+		if envVar, ok := environment["K6_COMPATIBILITY_MODE"]; ok {
+			opts.CompatibilityMode = null.StringFrom(envVar)
+		}
+	}
+	if _, err := lib.ValidateCompatibilityMode(opts.CompatibilityMode.String); err != nil {
+		// some early validation
+		return opts, err
+	}
+
+	if !opts.IncludeSystemEnvVars.Valid { // If not explicitly set via CLI flags, look for an environment variable
+		if envVar, ok := environment["K6_INCLUDE_SYSTEM_ENV_VARS"]; ok {
+			val, err := strconv.ParseBool(envVar)
+			if err != nil {
+				return opts, err
+			}
+			opts.IncludeSystemEnvVars = null.BoolFrom(val)
+		}
+	}
+
```

Comment:
I realize this fixes the handling of these env vars, but the special treatment of these specific ones feels like we're adding more things to iron out in #883. Hopefully we can have a saner way of dealing with these fallback scenarios.

No action needed, just a comment. :)

Reply:
I have a vague idea how to implement that logic in a generic way, when we get to that... It's not as if the fallback/overriding logic is very complicated, I don't even consider this "special treatment" - it is all pretty straightforward "if this is not set at the top tier, try the bottom one"... 

So, I think we would be able to implement something to handle all of the different types and cases, that is relatively simple, generic, and easy to test, even with the limitations imposed by Go.